### PR TITLE
Issue #348: Adopt the changes in 'liberty-maven-plugin' 3.0

### PR DESCRIPTION
**Pull request by josephca**

### CODE REVIEW

File: dev/com.ibm.etools.maven.liberty.integration/src/com/ibm/etools/maven/liberty/integration/internal/LibertyMaven.java [hunk 1]
```diff
@@ -187,7 +187,7 @@ public boolean isSupportedProject(IProject project, IProgressMonitor monitor) {
     /** {@inheritDoc} */
     @Override
     public void updateSrcConfig(IPath location, LibertyBuildPluginConfiguration config, IProgressMonitor monitor) {
-        String goal = "package liberty:install-apps";
+        String goal = "package liberty:deploy";
```
Comment (cherylking) [inline dev/com.ibm.etools.maven.liberty.integration/src/com/ibm/etools/maven/liberty/integration/internal/LibertyMaven.java]: Don't you need to check which version of the liberty maven plugin is configured before deciding whether to call install-apps or deploy goal? Before 3.x, install-apps was the correct goal to call. Calling deploy instead would not work if a 2.x version of the plugin was configured.

**Review by cherylking — CHANGES_REQUESTED** [review]
I'm not sure where the parameters for the goals are handled, but install-apps and deploy have different parameters that they expect. See the release notes for details: https://github.com/OpenLiberty/ci.maven/releases#changed-deploy-goal-and-removed-install-apps-goal

File: dev/com.ibm.etools.maven.liberty.integration/src/com/ibm/etools/maven/liberty/integration/internal/LibertyMaven.java (hunk 1 above)
Comment (c-koell) [reply dev/com.ibm.etools.maven.liberty.integration/src/com/ibm/etools/maven/liberty/integration/internal/LibertyMaven.java]: Please consider also the other changes like create-server -> create and so on ...

**Comment by josephca** [comment]
> I'm not sure where the parameters for the goals are handled, but install-apps and deploy have different parameters that they expect. See the release notes for details: https://github.com/OpenLiberty/ci.maven/releases#changed-deploy-goal-and-removed-install-apps-goal



> I'm not sure where the parameters for the goals are handled, but install-apps and deploy have different parameters that they expect. See the release notes for details: https://github.com/OpenLiberty/ci.maven/releases#changed-deploy-goal-and-removed-install-apps-goal

 In the existing 'install-apps' goal calls, no param has been passed, thus we don't need to map the parameters.

File: dev/com.ibm.etools.maven.liberty.integration/src/com/ibm/etools/maven/liberty/integration/internal/LibertyMaven.java (hunk 1 above)
Comment (josephca) [reply dev/com.ibm.etools.maven.liberty.integration/src/com/ibm/etools/maven/liberty/integration/internal/LibertyMaven.java]: > Please consider also the other changes like create-server -> create and so on ...


Yes, they will be handled together.

Comment (josephca) [reply dev/com.ibm.etools.maven.liberty.integration/src/com/ibm/etools/maven/liberty/integration/internal/LibertyMaven.java]: > Don't you need to check which version of the liberty maven plugin is configured before deciding whether to call install-apps or deploy goal? Before 3.x, install-apps was the correct goal to call. Calling deploy instead would not work if a 2.x version of the plugin was configured.

This update will result in some impact on Websphere Developer Tools as well, so handling both 2.x/3.x may not be an easy option. Rather, we should consider to document it to use 3.x for upcoming OLT/WDT releases (20.0.0.3 and higher)

**Pull request closed without merging**